db/db.go

```diff
@@ -3,19 +3,23 @@ package db
 import . "github.com/tendermint/tmlibs/common"
 
 type DB interface {
-	Get([]byte) []byte
+	Get([]byte) []byte // NOTE: returns nil iff never set or deleted.
 	Set([]byte, []byte)
 	SetSync([]byte, []byte)
 	Delete([]byte)
 	DeleteSync([]byte)
 	Close()
 	NewBatch() Batch
 	Iterator() Iterator
-	IteratorPrefix([]byte) Iterator
 
 	// For debugging
 	Print()
+
+	// Stats returns a map of property values for all keys and the size of the cache.
 	Stats() map[string]string
+
+	// CacheWrap wraps the DB w/ a CacheDB.
```

Comment:
Why does this return interface{}? It provides no guarantees.

In cachedb_test.go, I see this line often:
`cdb := mem.CacheWrap().(*CacheDB)`

If the only way to use it is to cast it to a particular type, then just force that type.
If you want people to be able to use the CacheWrap without forcing a given type, then return a usable interface,that exposes some subset of the DB interface.

Reply:
Actually, it seems *CacheDB implements the entire DB interface, why not return it there? Or remove a method or two that you don't want to require from Caches...

```Go
type Cache interface {
.....
}

type DB interface {
  Cache
  NewBatch() Batch
  Iterate() Iterator
}
```

then we could just use `cdb := mem.CacheWrap()`

Reply:
Maybe.  But CacheWrap also exists in the SDK for the merkle stores, and those don't conform to tmlibs/db.DB...